pkg/api/v1/types.go

```diff
@@ -611,7 +611,7 @@ type HTTPGetAction struct {
 	// Optional: Path to access on the HTTP server.
 	Path string `json:"path,omitempty" description:"path to access on the HTTP server"`
 	// Required: Name or number of the port to access on the container.
-	Port util.IntOrString `json:"port" description:"number or name of the port to access on the container"`
+	Port util.IntOrString `json:"port" description:"number or name of the port to access on the container; number must be in the range 1 to 65536; name must be compatible with IANA Service Name; see https://tools.ietf.org/html/rfc6335#section-5.1"`
```

Comment:
I know this is growing a bit, but I think we should promote this to the same level as DNS_LABEL.

Specifically:
1. Add a comment at the top of API files defining PORT_NAME (or maybe IANA_SVC_NAME?), documenting the regex and URL
2. Make comments like this reference PORT_NAME
3. Make ContainerPort.Name follow the same validation (no point being able to name a port that you can't refer to)

We have a slight problem: This RFC allows upper-case.   None of our usage sites handle that properly (e.g. the DNS server will assume it can use this name directly, but I think it has to normalize to lower-case).  How bad would it be if we only allowed lowercase?